src/lib/FreeC/LiftedIR/Syntax/Expr.hs

```diff
@@ -114,8 +111,9 @@ data ConPat = ConPat
 --   The variable pattern can optionally have a type signature.
 data VarPat = VarPat
   { varPatSrcSpan   :: SrcSpan
-  , varPatIdent     :: String
+  , varPatIdent     :: QName
```

Comment:
Neither qualified names nor symbols are allowed in variable patterns. Thus, I don't think that we should allow `QName`s here.